frontend/server/src/test/java/org/pytorch/serve/WorkflowTest.java

```diff
@@ -0,0 +1,507 @@
+package org.pytorch.serve;
+
+import io.netty.channel.Channel;
+import io.netty.handler.codec.http.DefaultFullHttpRequest;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.netty.handler.codec.http.HttpHeaderValues;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.HttpUtil;
+import io.netty.handler.codec.http.HttpVersion;
+import io.netty.util.CharsetUtil;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+import io.netty.util.internal.logging.Slf4JLoggerFactory;
+import java.io.File;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.concurrent.CountDownLatch;
+import org.apache.commons.io.FileUtils;
+import org.pytorch.serve.http.ErrorResponse;
+import org.pytorch.serve.http.StatusResponse;
+import org.pytorch.serve.servingsdk.impl.PluginsManager;
+import org.pytorch.serve.snapshot.InvalidSnapshotException;
+import org.pytorch.serve.util.ConfigManager;
+import org.pytorch.serve.util.ConnectorType;
+import org.pytorch.serve.util.JsonUtils;
+import org.pytorch.serve.workflow.messages.DescribeWorkflowResponse;
+import org.pytorch.serve.workflow.messages.ListWorkflowResponse;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+public class WorkflowTest {
```

Comment:
Add negative test cases
* a single prediction node failure in a multi node sequential workflow